### PR TITLE
Acrolinx 6/19

### PR DESCRIPTION
**Pull request by gewarren**

### CODE REVIEW

File: docs/csharp/how-to/modify-string-contents.md
Comment (tdykstra) [inline docs/csharp/how-to/modify-string-contents.md:9]:
```suggestion
This article demonstrates several techniques to produce a `string` by modifying an existing `string`. All the techniques demonstrated return the result of the modifications as a new `string` object. To demonstrate that the original and modified strings are distinct instances, the examples store the result in a new variable. You can examine the original `string` and the new, modified `string` when you run each example.
```

File: docs/csharp/how-to/parse-strings-using-split.md
Comment (tdykstra) [inline docs/csharp/how-to/parse-strings-using-split.md:3]: For SEO I think that String.Split would be better.

File: docs/csharp/how-to/parse-strings-using-split.md
Comment (tdykstra) [inline docs/csharp/how-to/parse-strings-using-split.md:41]:
```suggestion
The following example uses spaces, commas, periods, colons, and tabs as separating characters, which are passed to <xref:System.String.Split%2A> in an array .
```

File: docs/core/testing/unit-testing-best-practices.md
Comment (BillWagner) [inline docs/core/testing/unit-testing-best-practices.md:54]: Consider adding the adverb "often" here. I think that was the original intent:

```suggestion
The term *mock* is unfortunately often misused when talking about testing. The following points define the most common types of *fakes* when writing unit tests:
```

File: docs/framework/debug-trace-profile/jitcompilationstart-mda.md
Comment (tdykstra) [inline docs/framework/debug-trace-profile/jitcompilationstart-mda.md:2]:
```suggestion
title: jitCompilationStart managed debugging assistant (MDA)
```

File: docs/csharp/how-to/compare-strings.md
Comment (BillWagner) [inline docs/csharp/how-to/compare-strings.md:101]: With these edits, all the other changes use present tense instead of future tense. Should this do the same?

```suggestion
Culture-sensitive comparisons are typically used to compare and sort strings input by users with other strings input by users. The characters and sorting conventions of these strings might vary depending on the locale of the user's computer. Even strings that contain identical characters might sort differently depending on the culture of the current thread. In addition, try this sample code locally on a Windows machine, and you get the following results:
```

File: docs/framework/debug-trace-profile/jitcompilationstart-mda.md
Comment (tdykstra) [inline docs/framework/debug-trace-profile/jitcompilationstart-mda.md:20]:
```suggestion
 Not all the assemblies the program depends on have been generated into native format. Or those assemblies that have been generated into native format aren't registered correctly.  
```

**Review by BillWagner — APPROVED** [review]
This looks great @gewarren 

I noticed a couple small things to consider. You can choose if you like those suggestions, and then :shipit: when ready.

File: docs/framework/network-programming/automatic-proxy-detection.md
Comment (tdykstra) [inline docs/framework/network-programming/automatic-proxy-detection.md:38]:
```suggestion
 Proxies are configured on a per-connectoid basis. A connectoid is an item in the network connection dialog. It can be a physical network device, such as a modem or Ethernet card, or a virtual interface, such as a VPN connection that runs over a network device. When a connectoid changes, the proxy detection algorithm is run again. For example, proxy detection is run when a wireless connection changes an access point or a VPN is enabled.
```

**Review by tdykstra — APPROVED** [review]
LGTM, with a few suggestions. I got halfway through and then saw that Bill reviewed, so no need to continue.

File: docs/core/testing/unit-testing-best-practices.md
Comment (Thraka) [inline docs/core/testing/unit-testing-best-practices.md:56]: This one was missed

```suggestion
*Fake* - A fake is a generic term that can be used to describe either a stub or a mock object. Whether it's a stub or a mock depends on the context in which it's used. So in other words, a fake can be a stub or a mock.
```

File: docs/framework/network-programming/automatic-proxy-detection.md
Comment (Thraka) [inline docs/framework/network-programming/automatic-proxy-detection.md:27]: ❗ ❗ ❗ 
I wouldn't change anything related to Internet Explorer here without confirmation from the product group. **This technology is old and is probably coded for IE specifically**. When a statement like one changed here mentions that it checks IE directly, it's probably detecting IE directly and not trying to figure out the specific browser the user has installed as default.

I would revert this file.

File: docs/standard/design-guidelines/choosing-between-class-and-struct.md
Comment (Thraka) [inline docs/standard/design-guidelines/choosing-between-class-and-struct.md:20]: @BillWagner What is the deal with these articles? Since they are part of the design guideline book we're reprinting with permission, are we allowed to update these? I thought we weren't.

**Review by Thraka — CHANGES_REQUESTED** [review]
Mostly minor but two very important things we need to consider.

File: docs/csharp/how-to/compare-strings.md
Comment (gewarren) [reply docs/csharp/how-to/compare-strings.md:101]: Maybe, but because this sentence doesn't start with "if", the suggestion doesn't sound quite right to me.

File: docs/framework/debug-trace-profile/jitcompilationstart-mda.md
Comment (gewarren) [reply docs/framework/debug-trace-profile/jitcompilationstart-mda.md:20]: Without the comma, it seems like a sentence fragment to me. Is it correct to write a sentence like that?

File: docs/standard/design-guidelines/choosing-between-class-and-struct.md
Comment (gewarren) [reply docs/standard/design-guidelines/choosing-between-class-and-struct.md:20]: Will wait on an answer here before merging. I was wondering about this too, although I have updated these files in the past.

File: docs/framework/debug-trace-profile/jitcompilationstart-mda.md
Comment (tdykstra) [reply docs/framework/debug-trace-profile/jitcompilationstart-mda.md:20]: Here's one article about it https://lifehacker.com/you-can-start-a-sentence-with-and-but-or-or-1832984839

Comment (gewarren) [reply docs/framework/debug-trace-profile/jitcompilationstart-mda.md]: Interesting, thanks. It still read weirdly to me, so I changed the format to what the API docs use for exceptions. Does that look okay?

Comment (tdykstra) [reply docs/framework/debug-trace-profile/jitcompilationstart-mda.md]: Yes, that's fine. 😄

File: docs/standard/design-guidelines/choosing-between-class-and-struct.md
Comment (gewarren) [reply docs/standard/design-guidelines/choosing-between-class-and-struct.md]: Consensus is we should not edit anything except metadata.

**Pull request merged**
